### ISSUE TITLE
Is it possible for libtpa to take dpdk xdp as driver?

### ISSUE THREAD

**Issue by SoonyangZhang** [bytedance/libtpa#1]
Hello:
  Is it possible for libtpa to take dpdk xdp as driver?  With XDP, dpdk app can also coexist with host stack.

**Comment by yuanhanliu** [comment]
Yes, it is. And we are working on it. It should come out soon.

**Comment by SoonyangZhang** [comment]
I have got packets by testing dpdk-testpmd (https://doc.dpdk.org/dts/test_plans/af_xdp_test_plan.html) with a virtio_net device.
In libtpa, I build dpdk with af xdp lib. 
But when I test it with techo, I got an error:   EAL:failed to parse device net_af_xdp0.
It seems the executable file is not linked with librte_af_xdp.a.

**Comment by yuanhanliu** [comment]
Right. The pmd is not linked because it's not compiled: we used to compile mlx5 pmd only as that's the only one we support. You could check the **next** branch though: I have changed the dpdk build, to let it build all pmds by default.

Missing xdpf driver is one thing. The most important part is to write an ebpf program to steer some flows to libtpa. We have finished the initial work. I just haven't got the time to do some final cleanups to make it public. Hopefully, I could make it by the end of this week so that you can try it out.

**Comment by SoonyangZhang** [comment]
With cmake, I link libtpa with net_xdp, but when a tcp client sends syn, libtpa can send syn + ack. But the three handshake can not be established.  The syn + ack packet seems dropped by kernel (tcp client).

**Comment by yuanhanliu** [comment]
It's now pushed. You might want to have a try. FYI, you can find a quick guide at doc/howto_xdp.rst

**Comment by SoonyangZhang** [comment]
nice job
